### PR TITLE
[Program Migration] UI Updates

### LINKED ISSUE #7912: [Program Migration] Re-write frontend to match mocks
Rather than having program export and import in separate tabs, we want to incorporate them into the existing program edit options. Specifications can be found in these mocks: https://www.figma.com/design/QCBBEkf4nNKqUvTCblad8T/Admin---Program-applications?node-id=718-3768&t=44y4Td9PFW7Bb2Tx-1

### PR DESCRIPTION
**Pull request by rockycodes**
### Description

This PR makes changes to the UI for program import/export so that it matches the [designs](https://www.figma.com/design/QCBBEkf4nNKqUvTCblad8T/Admin---Program-applications?node-id=1-19&t=8deU4AZ3g29nvvFK-0) as closely as possible.

In the new flow, admin can export a program by click the "more actions" menu on a specific program and then selecting "Export program". They can import a program by clicking the "Import an existing program" link on the program index page.

See this screen recording for the full flow and some error states:

https://github.com/user-attachments/assets/54445e8e-d40d-4d69-9199-ca6c4388ce40

#### General

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

### Instructions for manual testing

1. Enable the `PROGRAM_MIGRATION_ENABLED` flag
2. Test out the import/export flow

### Issue(s) this completes

Fixes #7912

### CODE REVIEW

File: browser-test/src/admin/admin_program_migration.test.ts [hunk 1]
```diff
@@ -266,12 +258,63 @@ test.describe('program migration', () => {
...
+        'Your program has been successfully imported',
+      )
+      await validateScreenshot(page, 'saved-program-success')
+    })
+
+    await test.step('import another program and go directly to program edit page', async () => {
+      await adminProgramMigration.clickButton('Import another program')
+      await adminProgramMigration.expectImportPage()
+      await expect(page.getByRole('textbox')).toHaveValue('')
+      // replace the question admin name to avoid draft collision
```
Comment (dkatzz) [inline browser-test/src/admin/admin_program_migration.test.ts:305]: **suggestion (non-blocking):** You could consider making this and the comment below a test.step too

File: server/app/views/admin/programs/ProgramIndexView.java [hunk 2]
```diff
@@ -665,4 +708,14 @@ private ButtonTag renderManageProgramAdminsLink(ProgramDefinition program) {
             .withClass(ButtonStyles.CLEAR_WITH_ICON_FOR_DROPDOWN);
     return asRedirectElement(button, adminLink);
   }
+
+  private ButtonTag renderExportProgramLink(ProgramDefinition program) {
+    String adminLink = routes.AdminExportController.index(program.id()).url();
+    ButtonTag button =
+        makeSvgTextButton("Export program", Icons.DOWNLOAD)
+            // maybe don't need this id?
```
Comment (dkatzz) [inline server/app/views/admin/programs/ProgramIndexView.java:716]: **suggestion:** Seems like a legacy comment?

File: server/app/assets/javascripts/admin_import_view.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// import {assertNotNull} from './util'
```
Comment (dkatzz) [inline server/app/assets/javascripts/admin_import_view.ts:1]: **suggestion:** Remove comment

**Review by dkatzz — APPROVED** [review]
This is looking so good!!

File: server/app/assets/javascripts/admin_import_view.ts (hunk 3 above)
Comment (rockycodes) [reply server/app/assets/javascripts/admin_import_view.ts]: oh whoops. i had a commit waiting in the wings for this that i forgot to push up.

File: server/app/views/admin/programs/ProgramIndexView.java (hunk 2 above)
Comment (rockycodes) [reply server/app/views/admin/programs/ProgramIndexView.java]: removed this comment and the id.

File: browser-test/src/admin/admin_program_migration.test.ts (hunk 1 above)
Comment (rockycodes) [reply browser-test/src/admin/admin_program_migration.test.ts]: done!

**Pull request merged**